tests/js/arrows/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -1934,6 +1936,845 @@ export const bem = block =>
+);
+
+foo(action => action => action);
+
+=====================================output=====================================
+const fn1 = a => 3;
+const fn2 = a => b => 3;
+const fn3 = a => b => c => 3;
+const fn4 = a => b => c => d => 3;
+const fn5 = a => b => c => d => e => 3;
+const fn6 = a => b => c => d => e => g => 3;
+const fn7 = a => b => c => d => e => g => f => 3;
+
+const fn8 = a => ({ foo: bar, bar: baz, baz: foo });
+const fn9 = a => b => ({ foo: bar, bar: baz, baz: foo });
+const fn10 = a => b => c => ({ foo: bar, bar: baz, baz: foo });
+const fn11 = a => b => c => d => ({ foo: bar, bar: baz, baz: foo });
+const fn12 = a => b => c => d => e => ({ foo: bar, bar: baz, baz: foo });
+const fn13 = a => b => c => d => e => g => ({ foo: bar, bar: baz, baz: foo });
+const fn14 =
+  a =>
+  b =>
+  c =>
+  d =>
+  e =>
+  g =>
+  f =>
+  ({ foo: bar, bar: baz, baz: foo });
+
+const curryTest =
+  argument1 =>
+  argument2 =>
+  argument3 =>
+  argument4 =>
+  argument5 =>
+  argument6 =>
+  argument7 =>
+  argument8 =>
+  argument9 =>
+  argument10 =>
+  argument11 =>
+  argument12 =>
+  ({
+    foo: argument1,
+    bar: argument2,
+  });
+
+let curryTest2 =
+  argument1 =>
+  argument2 =>
+  argument3 =>
+  argument4 =>
+  argument5 =>
+  argument6 =>
+  argument7 =>
+  argument8 =>
+  argument9 =>
+  argument10 =>
+  argument11 =>
+  argument12 => {
+    const foo = "foo";
+    return foo + "bar";
+  };
+
+curryTest2 =
+  argument1 =>
+  argument2 =>
+  argument3 =>
+  argument4 =>
+  argument5 =>
+  argument6 =>
+  argument7 =>
+  argument8 =>
+  argument9 =>
+  argument10 =>
+  argument11 =>
+  argument12 => {
+    const foo = "foo";
+    return foo + "bar";
+  };
+
+throw argument1 =>
+  argument2 =>
+  argument3 =>
+  argument4 =>
+  argument5 =>
+  argument6 =>
+  argument7 =>
+  argument8 =>
+  argument9 =>
+  argument10 =>
+  argument11 =>
+  argument12 => {
+    const foo = "foo";
+    return foo + "bar";
+  };
+
+foo(
+  argument1 =>
+    argument2 =>
+    argument3 =>
+    argument4 =>
+    argument5 =>
+    argument6 =>
+    argument7 =>
+    argument8 =>
+    argument9 =>
+    argument10 =>
+    argument11 =>
+    argument12 =>
+    3
+);
+
+foo(
+  argument1 =>
+    argument2 =>
+    argument3 =>
+    argument4 =>
+    argument5 =>
+    argument6 =>
+    argument7 =>
+    argument8 =>
+    argument9 =>
+    argument10 =>
+    argument11 =>
+    argument12 =>
+    ({
+      foo: bar,
+      bar: baz,
+      baz: foo,
+    })
+);
+
+foo(
+  argument1 =>
+    argument2 =>
+    argument3 =>
+    argument4 =>
+    argument5 =>
+    argument6 =>
+    argument7 =>
+    argument8 =>
+    argument9 =>
+    argument10 =>
+    argument11 =>
+    argument12 => {
+      const foo = "foo";
+      return foo + "bar";
+    }
+);
+
+(
+  argument1 =>
+  argument2 =>
+  argument3 =>
+  argument4 =>
+  argument5 =>
+  argument6 =>
+  argument7 =>
+  argument8 =>
+  argument9 =>
+  argument10 =>
+  argument11 =>
+  argument12 =>
+  3
+)(3);
```

Comment:
Can you add indention for this? I think this is `CallExpression.callee`?

Reply:
like below?
```js
(
  argument1 =>
    argument2 =>
    argument3 =>
    argument4 =>
    argument5 =>
    argument6 =>
    argument7 =>
    argument8 =>
    argument9 =>
    argument10 =>
    argument11 =>
    argument12 =>
    3
)(3);
```

Reply:
Yes, WDYT?

Reply:
It was intended to be unified with the format for binary expressions. But I have no strong opinion.

**Prettier pr-9992**
[Playground link](https://deploy-preview-9992--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAKAOlABF1AzCQoiASiwF4A+THfYosqm3A+0i67Ft9pruto061W9Ic1FtMJVCQDcmTBn6SGWANTMBYjVtWEymlT0N6Tu44Is59RabIVQQAGhAQADjACW0AM7JQAEMAJ2CIAHcABRCEfxRAgBtwwIBPf1cAI2DAsABrOBgAZXccrygAc2QYYIBXOFc4AFsMuAATVraAGUCKmsDyuAAxCGDGwJhvCuQQQJqYCBcQAAsYRoSAdSWveF8SsDhC2O2vADdtlOmwX3SQMt84YJhI7PKx5DxE+9cAK18ADwAQtk8gVCoFGnBOmU4O9PvUQL8-oUyuUEnAAIo1CDwWEJL4gErBe7BaYZQItBKLdzBMowdZeVowJbIAAcAAZXNSIPd1tl3NNqXBiScYa4AI5Y+DPDxxGa+AC0UDgbTai2CcAlXnVz36byQHzx8PujS8VVqRpRaMx2Jh+rhrhg5PpjOZSAATA7sl4EiiAMIQRp6kBCgCsixq9wAKuS4gb8Sc6gBJKAdWCFMA0zwAQRThRgKTRuPuAF9i0A)
```sh
--parser babel
```

**Input:**
```jsx
(
  (fooooooo) =>
  (fooooooo) =>
  (fooooooo) =>
  (fooooooo) =>
  (fooooooo) =>
  (fooooooo) =>
  fooooooooo
)();

(
  (fooooooo) +
  (fooooooo) +
  (fooooooo) +
  (fooooooo) +
  (fooooooo) +
  (fooooooo) +
  fooooooooo
)();

```

**Output:**
```jsx
(
  (fooooooo) =>
  (fooooooo) =>
  (fooooooo) =>
  (fooooooo) =>
  (fooooooo) =>
  (fooooooo) =>
  fooooooooo
)();

(
  fooooooo +
  fooooooo +
  fooooooo +
  fooooooo +
  fooooooo +
  fooooooo +
  fooooooooo
)();

```